### PR TITLE
GUI: Refactor common UI elements into a single module.

### PR DESCRIPTION
**Pull request by russellhancox**
At the same time, update the FAA dialog to match the new style

FAA main view:
image

FAA "More Details":
image

### CODE REVIEW

**Comment by pmarkowsky** [comment]
Main concern here is that I'd want the process and username that performed the file access rather than the rule meta data in the main part of the FAA window.

File: Source/common/SNTBlockMessage.h [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /// Copyright 2016 Google Inc. All rights reserved.
+/// Copyright 2014 North Pole Security, Inc.
```
Comment (pmarkowsky) [inline Source/common/SNTBlockMessage.h:2]:
```suggestion
/// Copyright 2024 North Pole Security, Inc.
```

File: Source/common/SNTFileAccessEvent.h [hunk 2]
```diff
@@ -37,17 +38,4 @@
...
-///
-@property(readonly) NSString *publisherInfo;
-
-///
-/// Return an array of the underlying SecCertificateRef's of the signingChain
-///
-/// WARNING: If the refs need to be used for a long time be careful to properly
-/// CFRetain/CFRelease the returned items.
-///
-@property(readonly) NSArray *signingChainCertRefs;
```
Comment (pmarkowsky) [inline Source/common/SNTFileAccessEvent.h:51]: Don't we need this for the process that caused the file access policy violation?

Comment (russellhancox) [reply Source/common/SNTFileAccessEvent.h:51]: I've moved it into the superclass

**Comment by russellhancox** [comment]
> Main concern here is that I'd want the process and username that performed the file access rather than the rule meta data in the main part of the FAA window.

I've added user (which was never previously a thing in this UI). 

Process is ambiguous: the PID is never shown (nor is it for binary blocks) but the process name is, and the parent is shown in the more details window.

**Review by tburgin — APPROVED** [review]

**Pull request merged**